src/test/java/com/fasterxml/jackson/core/io/BigDecimalParserTest.java

```diff
@@ -0,0 +1,18 @@
+package com.fasterxml.jackson.core.io;
+
+public class BigDecimalParserTest extends com.fasterxml.jackson.core.BaseTest {
+    public void testLongStringParse() {
+        final int len = 1500;
+        final StringBuilder sb = new StringBuilder(len);
+        for (int i = 0; i < len; i++) {
+            sb.append("A");
+        }
+        try {
+            BigDecimalParser.parse(sb.toString());
+            fail("expected NumberFormatException");
+        } catch (NumberFormatException nfe) {
+            assertTrue("exception message starts as expected?", nfe.getMessage().startsWith("Value \"AAAAA"));
```

Comment:
There is also helper method `verifyException` in test base class for simple inclusion checks.
(this is fine too)